src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,53 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final String DATE_PATTERN = "dd.MM.yyyy";
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_PATTERN);
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOURS_WORKED_INDEX = 2;
+    private static final int HOURLY_RATE_INDEX = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        if (names == null || data == null || dateFrom == null || dateTo == null) {
+            throw new IllegalArgumentException("Arguments cannot be null");
+        }
+        StringBuilder salaryInformation = new StringBuilder("Report for period "
+                + dateFrom + " - " + dateTo);
+        for (String name : names) {
+            int totalSalary = 0;
+            salaryInformation.append(System.lineSeparator());
```

Comment:
Instead of appending `System.lineSeparator()` directly to the `StringBuilder`, it is more efficient to include it as part of the subsequent append operations to reduce the number of append calls.